Enable Tailscale's Funnel feature (#197)
hassio-addons/addon-tailscale: tailscale/DOCS.md

Reviewer: Please add the trusted proxy information in this documentation

Author: I've added a short reference from funnel docs to proxy docs.

When not set, this option is enabled by default.

### Option: `funnel`

This requires Tailscale Proxy to be enabled.

**Important:** See also the "Option: `proxy`" section of this documentation for the
necessary configuration changes in Home Assistant!

When not set, this option is enabled by default.

> With the Tailscale Funnel feature you can access your Home Assistant instance
> from the wider internet using your Tailscale domain (like
> `https://homeassistant.tail1234.ts.net`) even from devices **without installed
> Tailscale VPN client** (eg. general phones, tablets, laptops).

Reviewer 2: ```suggestion
With the Tailscale Funnel feature, you can access your Home Assistant instance
from the wider internet using your Tailscale domain (like
`https://homeassistant.tail1234.ts.net`) even from devices **without installed
Tailscale VPN client** (for example, on general phones, tablets, and laptops).
```

**Client** &#8658; _Internet_ &#8658; **Tailscale Funnel** (TCP proxy) &#8658;
_VPN_ &#8658; **Tailscale Proxy** (HTTPS proxy) &#8594; **HA** (HTTP web-server)

> Without the Tailscale Funnel feature, you will be able to access your Home
> Assistant instance only when your devices (eg. phones, tablets, laptops) are
> connected to your Tailscale VPN, there will be no Internet &#8658; VPN TCP
> proxying for HTTPS communication.

Reviewer 2: ```suggestion
Without the Tailscale Funnel feature, you will be able to access your Home
Assistant instance only when your devices (for example, phones, tablets, and laptops)
are connected to your Tailscale VPN, there will be no Internet &#8658; VPN TCP
proxying for HTTPS communication.
```

More information: [Tailscale Funnel][tailscale_info_funnel]

1. Navigate to the [Access controls page][tailscale_acls] of the admin console,
   and add the below policy entries to the policy file. See [Server role
   accounts using ACL tags][tailscale_info_acls] for more information.

   ```json
   {
     "nodeAttrs": [
       {
         "target": ["autogroup:members"],
         "attr": ["funnel"]
       }
     ]
   }
   ```

1. Restart the add-on.

> **Note**: _After initial set up it can take up to 10 minutes for the domain to
> be publicly available. You can use the `dig` command (Linux/MacOS) to regularly
> check if an A-record is already present for your domain (`dig
> <machine-name>.<tailnet-name>.ts.net +short` should return an IP address once
> the record is published)._

Reviewer 2: Unneeded extra information, will generally only cause confusion with newer users.
Expert users will know their way around dig and similar tools.

```suggestion
**Note**: _After initial setup, it can take up to 10 minutes for the domain to
be publicly available._
```

> **Note:** _You should not use any port number in the url that you used
> previously to access Home Assistant. Tailscale Funnel works on the default HTTPS
> port 443._

Reviewer 2: ```suggestion
**Note:** _You should not use any port number in the URL that you used
previously to access Home Assistant. Tailscale Funnel works on the default HTTPS
port 443._
```
